app/src/marketplace/modules/dataUnion/services.js

```diff
@@ -556,39 +556,111 @@ const getSidechainEvents = async (address: string, eventName: string, fromBlock:
     return events
 }
 
-export const getJoinsAndParts = async (address: string, fromTimestamp: number) => {
+export const getJoinsAndParts = async (id: DataUnionId, fromTimestamp: number) => {
     const web3 = getSidechainWeb3()
     const fromBlock = await getBlockNumberForTimestamp(web3, Math.floor(fromTimestamp / 1000))
-    const joins = await getSidechainEvents(address, 'MemberJoined', fromBlock)
-    const parts = await getSidechainEvents(address, 'MemberParted', fromBlock)
+    const joinEvents = await getSidechainEvents(id, 'MemberJoined', fromBlock)
+    const partEvents = await getSidechainEvents(id, 'MemberParted', fromBlock)
```

Comment:
It's from earlier code but `getSideChainEvents` calls `getSidechainContract` which has:

```js
const web3 = getSidechainWeb3()
return new web3.eth.Contract(getConfig().dataUnionAbi, address)
```

The `dataUnionAbi` stored in the frontend is the old DU1 contract abi, the DU2 contract code is not stored since everything is handled in the client. Seems like this works by coincidence... maybe the event definitions haven't changed there but there is change something breaking here in the future.

Reply:
Oh, good catch! I'll check the ABI.

Reply:
If we are only interested in one event, we could just define a partial abi like in the client: https://github.com/streamr-dev/streamr-client-javascript/blob/21d668d228c560a0569908c19329220f0235b2a7/src/dataunion/Contracts.ts#L181